sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1431,4 +1484,30 @@ static void validateTableExists(BigtableConfig config, PipelineOptions options)
       }
     }
   }
+
+  interface ResourceStatsSupplier {
+    ResourceLimiterStats getStats();
+  }
+
+  private static class ResourceStatsSupplierImpl implements ResourceStatsSupplier, Serializable {
+
+    private final BigtableConfig bigtableConfig;
+
+    ResourceStatsSupplierImpl(BigtableConfig bigtableConfig) {
+      this.bigtableConfig = bigtableConfig;
+    }
+
+    @Override
+    public ResourceLimiterStats getStats() {
+      if (bigtableConfig.getProjectId() == null || bigtableConfig.getInstanceId() == null) {
+        return ResourceLimiterStats.getInstance(
+            new BigtableInstanceName(
+                bigtableConfig.getBigtableOptions().getProjectId(),
+                bigtableConfig.getBigtableOptions().getInstanceId()));
+      }
+      return ResourceLimiterStats.getInstance(
+          new BigtableInstanceName(
+              bigtableConfig.getProjectId().get(), bigtableConfig.getInstanceId().get()));
+    }
+  }
```

Comment:
I think it would more clear to factor out the instantiation:
String projectId = bigtableConfig.getProjectId();
if (projectId == null) {
  projectId = bigtableConfig.getBigtableOptions().getProjectId();
}
...
InstanceName name = new BigtableInstanceName(projectId...);
return ResourceLimiterStats.getInstance(name);

Reply:
Included instanceId just in case, done.